File: OnlineShop.Client/src/model/Product.ts
import { Discount } from './Discount';
import { ProductCategory } from './ProductCategory';

export type Product = {
  id: number;
  name: string;
  productCategory: ProductCategory;
  discount: Discount;
  description: string;
  price: number;
  quantity: number;
  sku: string;
};

File: OnlineShop.Core/Model/Product.cs
namespace OnlineShop.Core.Model;

public class Product : BaseEntity
{
    public required string Name { get; set; }
    public int DiscountId { get; set; }
    public int ProductCategoryId { get; set; }
    public required string Description { get; set; }
    public double Price { get; set; }
    public int Quantity { get; set; }
    public required string Sku { get; set; }

    public required Discount Discount { get; set; }
    public required ProductCategory ProductCategory { get; set; }
    public ICollection<Wishlist> Wishlists { get; set; } = new List<Wishlist>();
    public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();

    //public IEnumerable<ProductAttachment> ProductAttachments { get; set; }
}